File: server/Models/VaultKeep.cs
namespace keeper.Models;

public class VaultKeep : RepoItem<int>
{
    public int KeepId { get; set; }
    public int VaultId { get; set; }
    public string CreatorId { get; set; }
    // public Profile Creator { get; set; 
}


public class VaultKeepProfile : Profile
{
    public string VaultKeepId { get; set; }
    public int KeepId { get; set; }
}

public class VaultKeepKeep : Keep
{
    public int VaultKeepId { get; set; }
    public int VaultId { get; set; }
    public string AccountId { get; set; }
}

File: client/src/models/VaultKeep.js
import { Account } from "./Account.js";
import { Keep } from "./Keep.js";
import { Vault } from "./Vault.js";

export class VaultKeep {
  constructor(data) {
    this.id = data.id;
    this.vaultId = data.vaultId;
    this.keepId = data.keepId;
  }
}

export class VaultKeepKeep extends Keep {
  constructor(data) {
    super(data);
    this.vaultKeepId = data.vaultKeepId;
    this.vaultId = data.vaultId;
  }
}

export class VaultProfile extends VaultKeep {
  constructor(data) {
    super(data)
    this.profile = new Account(data.profile)
  }
}

export class VaultKeepVault extends VaultKeep {
  constructor(data) {
    super(data)
    this.vault = new Vault(data.album)
  }
}
